src/rdb.c

```diff
@@ -3499,6 +3517,20 @@ void killRDBChild(void) {
      * - rdbRemoveTempFile */
 }
 
+/* Send to replica End Offset response with structure
+ * $ENDOFF:<end-offset> <primary-repl-id> <current-db-id> */
+int sendCurentOffsetToReplica(client* replica) {
+    char buf[128];
+    int buflen;
+    buflen = snprintf(buf, sizeof(buf), "$ENDOFF:%lld %s %d\r\n", server.master_repl_offset, server.replid, server.db->id);
+    serverLog(LL_NOTICE, "Sending to replica %s RDB end offset %lld", replicationGetSlaveName(replica), server.master_repl_offset);    
+    if (connWrite(replica->conn, buf, buflen) != buflen) {
```

Comment:
Shouldn't this be like a sync write? It seems possible this might be a partial write.